lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -460,6 +460,24 @@ public function getRootAliases()
     }
 
     /**
+     * Gets all the aliases that have been used in the query.
+     * Including all select root aliases and join aliases
+     *
+     * <code>
+     *     $qb = $em->createQueryBuilder()
+     *         ->select('u')
+     *         ->from('User', 'u')
+     *         ->join('u.articles','a';
+     *
+     *     $qb->getAllAliases(); // array('u','a')
+     * </code>
+     * @return array
+     */
+    public function getAllAliases() {
```

Comment:
Would you consider modifying getRootAliases() to include all of the JOIN aliases instead of just the SELECT ones?